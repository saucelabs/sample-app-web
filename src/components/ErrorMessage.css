.error-message-container {
  background-color: #ffffff;
  height: 45px;
  margin-bottom: 5px;
  margin-top: -10px;
  position: relative;
  flex: 1;
  display: flex;
  align-items: center;
  padding-left: 10px;
  padding-right: 10px;
  justify-content: center;
}

.error-message-container.error {
  background-color: #e2231a;
}

.error-message-container h3 {
  color: #ffffff;
  font-size: 14px;
  text-align: center;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  padding-left: 10px;
  padding-right: 10px;
}

.error-message-container .error-button {
  border: 0px;
  background-color: transparent;
  color: #ffffff;
  cursor: pointer;
  position: absolute;
  right: 5px;
  top: 5px;
}

.error-message-container .error-button:focus {
  outline: none;
}
