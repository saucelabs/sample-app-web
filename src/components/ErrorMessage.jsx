import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTimes } from "@fortawesome/free-solid-svg-icons";
import "./ErrorMessage.css";

const ErrorMessage = ({ isError, errorMessage, onClick, ...props }) => {
  return (
    <div
      className={`error-message-container${isError ? " error" : ""}`}
      {...props}
    >
      {isError && (
        // This component is not structured how it should,
        // But this is done to keep backwards compatibility
        <h3 data-test="error">
          <button className="error-button" onClick={onClick}>
            <FontAwesomeIcon icon={faTimes} />
          </button>
          {errorMessage}
        </h3>
      )}
    </div>
  );
};

ErrorMessage.propTypes = {
  /**
   * If this is an isError field yes or no
   */
  isError: PropTypes.bool.isRequired,
  /**
   * The value of the input
   */
  errorMessage: PropTypes.string.isRequired,
  /**
   * The on change handler
   */
  onClick: PropTypes.func.isRequired,
};

export default ErrorMessage;
