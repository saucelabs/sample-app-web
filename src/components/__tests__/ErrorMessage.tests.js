import React from "react";
import { shallow } from "enzyme";
import ErrorMessage from "../ErrorMessage";

describe("ErrorMessage", () => {
  it("should render no error message when there is not error", () => {
    const props = {
      isError: false,
      errorMessage: "Error Message",
      onClick: () => {},
    };
    const component = shallow(<ErrorMessage {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should render an error message when there is an error", () => {
    const props = {
      isError: true,
      errorMessage: "Error Message",
      onClick: () => {},
    };
    const component = shallow(<ErrorMessage {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should be able to render custom props on the container level", () => {
    const props = {
      isError: false,
      errorMessage: "",
      onClick: () => {},
      foo: "bar",
      bar: "foo-bar",
    };
    const component = shallow(<ErrorMessage {...props} />);

    expect(component).toMatchSnapshot();
  });
});
