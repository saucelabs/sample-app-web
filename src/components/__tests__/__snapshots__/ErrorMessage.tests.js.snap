// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should be able to render custom props on the container level 1`] = `
<div
  bar="foo-bar"
  className="error-message-container"
  foo="bar"
/>
`;

exports[`ErrorMessage should render an error message when there is an error 1`] = `
<div
  className="error-message-container error"
>
  <h3
    data-test="error"
  >
    <button
      className="error-button"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              352,
              512,
              Array [],
              "f00d",
              "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
            ],
            "iconName": "times",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    Error Message
  </h3>
</div>
`;

exports[`ErrorMessage should render no error message when there is not error 1`] = `
<div
  className="error-message-container"
/>
`;
